ACE/ace/SOCK_Dgram_Mcast.h

```diff
@@ -1,407 +1,407 @@
-  int set_option (int option, char optval);
-
-  /// Dump the state of an object.
-  /**
-   * Logs the setting of all options, the bound address, the send address and
-   * interface, and the list of current subscriptions.
-   */
-  void dump () const;
-
-  /// Declare the dynamic allocation hooks.
-  ACE_ALLOC_HOOK_DECLARE;
-
-  /// Override write accessor for the constructor options (@see enum options above)
-  /// This class is typically default instantiated in a connection handler templated
-  /// framework so these cannot be specified on construction.
-  void opts (int opts);
-
-  /// Read accessor for the constructor options (@see enum options above)
-  int opts () const;
-
-private:
-
-  /// Subscribe to a multicast address on one or more network interface(s).
-  /// (No QoS support.)
-  int subscribe_ifs (const ACE_INET_Addr &mcast_addr,
-                     const ACE_TCHAR *net_if,
-                     int reuse_addr);
-
-  /// Do subscription processing w/out updating the subscription list.
-  /// (Layered method for subscribe() processing).
-  int subscribe_i (const ACE_INET_Addr &mcast_addr,
-                   int reuse_addr = 1,
-                   const ACE_TCHAR *net_if = 0);
-
-  /// Unsubscribe from a multicast address on one or more network interface(s).
-  int unsubscribe_ifs (const ACE_INET_Addr &mcast_addr,
-                       const ACE_TCHAR *net_if = 0);
-
-  /// Do unsubscription processing w/out udpating subscription list.
-  ///  (Layered method for <unsubscribe> processing).
-  int unsubscribe_i (const ACE_INET_Addr &mcast_addr,
-                     const ACE_TCHAR *net_if = 0);
-
-protected:
-  /// Contains common open functionality so that inheriting classes can
-  /// reuse it.
-  int open_i (const ACE_INET_Addr &mcast_addr,        // Bound & sendto address.
-              const ACE_TCHAR *net_if = 0,            // Net interface for sends.
-              int reuse_addr = 1);
-
-  /// Empty the dynamic subscription list.
-  int clear_subs_list ();
-
-private:
-  /// Per-instance options..
-  int opts_;
-
-  /// Multicast address to which local send() methods send datagrams.
-  ACE_INET_Addr  send_addr_;
-  /// Network interface to which all send() methods send multicast datagrams.
-  ACE_TCHAR *send_net_if_;
-
-#if defined (ACE_SOCK_DGRAM_MCAST_DUMPABLE)
- typedef ACE_DLList<ip_mreq>  subscription_list_t;
- typedef ACE_DLList_Iterator<ip_mreq>  subscription_list_iter_t;
- /// List of currently subscribed addr/iface pairs (and assc. types).
- mutable subscription_list_t  subscription_list_;
- /// Lock used to protect subscription list.
- mutable ACE_SDM_LOCK subscription_list_lock_;
-     // (Lock type does not need to support recursive locking.)
-#endif /* ACE_SOCK_DGRAM_MCAST_DUMPABLE */
-
-};
-
-ACE_END_VERSIONED_NAMESPACE_DECL
-
-#if defined (__ACE_INLINE__)
-#include "ace/SOCK_Dgram_Mcast.inl"
-#endif /* __ACE_INLINE__ */
-
-#include /**/ "ace/post.h"
-#endif /* ACE_SOCK_DGRAM_MCAST_H */
+// -*- C++ -*-

```

Comment:
Github shows this file as 100% changed, please check that, now it can't be reviewed

Reply:
I resubmitted the file